diff --git a/node_modules/inquirer-command-prompt/index.js b/node_modules/inquirer-command-prompt/index.js
index 9192101..6c1133f 100644
--- a/node_modules/inquirer-command-prompt/index.js
+++ b/node_modules/inquirer-command-prompt/index.js
@@ -46,7 +46,7 @@ class CommandPrompt extends InputPrompt {
   }
 
   async initAutoCompletion(context, autoCompletion) {
-    if (!autoCompleters[context]) {
+    // if (!autoCompleters[context]) {
       if (thiz.isAsyncFunc(autoCompletion)) {
         autoCompleters[context] = async l => this.asyncAutoCompleter(l, autoCompletion)
       } else if (autoCompletion) {
@@ -54,7 +54,7 @@ class CommandPrompt extends InputPrompt {
       } else {
         autoCompleters[context] = () => []
       }
-    }
+    // }
   }
 
   static addToHistory(context, value) {
@@ -394,4 +394,3 @@ class CommandPrompt extends InputPrompt {
 let thiz = CommandPrompt
 
 module.exports = CommandPrompt
-
